In [5]:
import pandas as pd
import numpy as np

#### First, I read file and clearn data

In [3]:
##### Read file aac_intakes.csv
aac_intakes = pd.read_csv(r'C:\\Users\\tuong\\Documents\\Python_Data_Analyst_Treehouse\\Project_5\\DataSet\\aac_intakes.csv')

##### Read file aac_outcomes.csv
aac_outcomes = pd.read_csv(r'C:\\Users\\tuong\Documents\\Python_Data_Analyst_Treehouse\\Project_5\DataSet\\aac_outcomes.csv')

##### Because we need the history data record, so we don't create an update dataframe
####Clearn the data####

# Convert object type to datetime64[ns] type in column datetime
aac_intakes[['datetime','datetime2']] = aac_intakes[['datetime','datetime2']].apply(pd.to_datetime)
aac_outcomes[['date_of_birth','datetime','monthyear']] = aac_outcomes[['date_of_birth','datetime','monthyear']].apply(pd.to_datetime)

# Clearn the duplicate row and nan in data
aac_intakes = aac_intakes.drop_duplicates()
aac_intakes = aac_intakes.dropna(subset = ['animal_id', 'datetime','datetime2'])

aac_outcomes = aac_outcomes.drop_duplicates()
aac_outcomes = aac_outcomes.dropna(subset = ['animal_id', 'date_of_birth','datetime','monthyear'])

#### My new answer

In [4]:
'''2.What is the average number of pets found in a month in the year 2015? 
Are there months where there is a higher number of animals found?'''

# Find the animal had found in 2015
found_2015 = aac_intakes[(aac_intakes['datetime'] > "2015-01-01") & (aac_intakes['datetime'] < "2015-12-31")]

# Find the highest number of animals had found in month in 2015 
Max_month_found_in_2015 = found_2015.groupby(found_2015.datetime.dt.month)['animal_id'].count().sort_values().max()

# The month had highest number of animals had found in 2015
# I add parameter 'ascending=False' to select the correct month name
month = found_2015.groupby(found_2015.datetime.dt.month_name())['animal_id'].count().sort_values(ascending=False).index[0]

print('The average number of pets found in every month in 2015: ', round(found_2015.shape[0]/12 ,2))
print(month, 'is the highest number of animals had found with: ', Max_month_found_in_2015)

The average number of pets found in every month in 2015:  1556.33
June is the highest number of animals had found with:  2189


In [7]:
'''5. What are the adoption rates for specific breeds?
Find the top 5 dog breeds in the shelter (based on count) and then find the adoption percentage of each breed.'''
# I want to check all history data so I use aac_intakes and aac_outcomes dataframe to calculate

## Step 1 ##
# Find rows have 'Dog' breed in shelter
breed_Dog_Shelter = aac_intakes[aac_intakes['animal_type'] == 'Dog']
# Find the top 5 'Dog' breed in shelter
top5_breedDog_Shelter = breed_Dog_Shelter.groupby(breed_Dog_Shelter['breed']).animal_id.count().sort_values(ascending=[False]).head(5)

## Step 2 ##
# Find the 'Dog' breed and 'Adoption' outcome_type in Update_aac_outcomes
Dog_outcomes_Adoption = aac_outcomes[(aac_outcomes['animal_type'] == 'Dog') & (aac_outcomes['outcome_type'] == 'Adoption')]
# Group data by 'breed' and count number 
breedDog_outcomes = Dog_outcomes_Adoption.groupby(Dog_outcomes_Adoption['breed']).animal_id.count()

## Step 3 ##
# List values for top 5 dogs in shelter
List1_number = []
for data_row in range(top5_breedDog_Shelter.shape[0]):
    values_data_row = top5_breedDog_Shelter.iloc[data_row]
    List1_number.append(values_data_row)
    
# List values find the number of top 5 dogs in shelter had record in breedDog_outcomes
List2_number = []
for breed in top5_breedDog_Shelter.index:
    data_row = breedDog_outcomes[breedDog_outcomes.index == str(breed)]
    values_data_row = data_row.iloc[0]
    List2_number.append(values_data_row)
    
# List for percentage of top 5 breed dog had adopted in shelter
List3_number = []  
for i in range(top5_breedDog_Shelter.shape[0]):
    number = round(100 / List1_number[i] * List2_number[i],2)
    List3_number.append(number)
    
# List for index use breed's dog data
List_Index = []
for i in top5_breedDog_Shelter.index:
    List_Index.append(i)

# Show the resault by create dataframe by 3 lists
display_Top5_percentage_breedDog = pd.DataFrame(list(zip(List_Index, List1_number, List2_number, List3_number)),
               columns =['Breed_Dog', 'Number_intakes', 'Number_adopted', 'Percentage'])
display_Top5_percentage_breedDog

Breed_Dog  Number_intakes  Number_adopted  Percentage
0               Pit Bull Mix            6379            2382       37.34
1    Chihuahua Shorthair Mix            4860            2293       47.18
2     Labrador Retriever Mix            4840            2404       49.67
3        German Shepherd Mix            1963             937       47.73
4  Australian Cattle Dog Mix            1105             619       56.02

- Final answer number 5:
The top 5 breeds dogs have 'Adopted' type is:
<br>Pit Bull Mix - 37.34%
<br>Chihuahua Shorthair Mix - 47.18%
<br>Labrador Retriever Mix - 49.67%
<br>German Shepherd Mix - 47.73%
<br>Australian Cattle Dog Mix - 56.02%


In [8]:
'''6. What are the adoption rates for different colorings?
Find the top 5 colorings in the shelter (based on count) and then find the adoption percentage of each color.'''
# I want to check all history data so I use aac_intakes and aac_outcomes dataframe to calculate

## Step 1 ##
# Find rows have 'Dog' breed in shelter
breed_Dog_Shelter = aac_intakes[aac_intakes['animal_type'] == 'Dog']
# Find the top 5 'Dog' breed in shelter
top5_colors_Dog_Shelter = breed_Dog_Shelter.groupby(breed_Dog_Shelter['color']).animal_id.count().sort_values(ascending=[False]).head(5)
top5_colors_Dog_Shelter

## Step 2 ##
# Find the 'Dog' breed and 'Adoption' outcome_type in Update_aac_outcomes
Dog_outcomes_Adoption = aac_outcomes[(aac_outcomes['animal_type'] == 'Dog') & (aac_outcomes['outcome_type'] == 'Adoption')]
# Group data by 'color' and count number 
colors_Dog_outcomes = Dog_outcomes_Adoption.groupby(Dog_outcomes_Adoption['color']).animal_id.count()
colors_Dog_outcomes

## Step 3 ##
# List values for top 5 color dog's in shelter
List1_number = []
for values in range(top5_colors_Dog_Shelter.shape[0]):
    values_data_row = top5_colors_Dog_Shelter.iloc[values]
    List1_number.append(values_data_row)

# List values find the number of top 5 dogs in shelter had record in breedDog_outcomes
List2_number = []
for values in top5_colors_Dog_Shelter.index:
    data_row = colors_Dog_outcomes[colors_Dog_outcomes.index == str(values)]
    values_data_row = data_row.iloc[0]
    List2_number.append(values_data_row)
    
# List for percentage of top 5 breed dog had adopted in shelter
List3_number = []  
for i in range(top5_colors_Dog_Shelter.shape[0]):
    number = round(100 / List1_number[i] * List2_number[i],2)
    List3_number.append(number)
    
# List for index use breed's dog data
List_Index = []
for i in top5_colors_Dog_Shelter.index:
    List_Index.append(i)
    
# Show the resault by create dataframe by 3 lists
display_Top5_percentage_colorsDog = pd.DataFrame(list(zip(List_Index, List1_number, List2_number, List3_number)),
               columns =['Color_dog', 'Number_intakes', 'Number_adopted', 'Percentage'])
display_Top5_percentage_colorsDog

Color_dog  Number_intakes  Number_adopted  Percentage
0  Black/White            5253            2609       49.67
1        Black            2527            1089       43.09
2  Brown/White            2500            1123       44.92
3    Tan/White            2473            1178       47.63
4        White            2378             932       39.19

- Final answer number 6: There are top 5 color have 'Adopted' type
<br>Black/White - 49.67%
<br>Black - 43.09%
<br>Brown/White - 44.92%
<br>Tan/White - 47.63%
<br>White - 39.19%

